### DISCUSSION TITLE
How to serialize enum to int

### DISCUSSION THREAD

**Discussion by GardenHamster** [aaubry/YamlDotNet#863] [Q&A] [answered]
``` csharp
public enum TestEnum
{
    TypeA=0, 
    TypeB=1, 
    TypeC=2, 
    TypeD=3,
}
```
``` csharp
public class TestClass
{
    public TestEnum Type { get; set; }
}
```
``` csharp
var data = new TestClass
{
    Type = TestEnum.TypeC
};
var serializer = new SerializerBuilder().Build();
var content = serializer.Serialize(data);
Console.WriteLine(content);
```

output
```
Type: TypeC
```

What I expect is
```
Type: 2
```

**Comment by EdwardCooke** [comment, marked as answer]
I don't think it does this out of the box, nor do I think there is an option for it. You could create a custom type converter for it though.

Here's a very basic example.

https://github.com/aaubry/YamlDotNet/blob/master/YamlDotNet/Serialization/Converters/GuidConverter.cs

In the accepts type you could do a `type.IsEnum` or something like that and write out an scalar with the int version of the enum in the WriteYaml and read in a Scalar as the int and parse it through Enum.TryParse (or something like that) and return that value.

**Comment by GardenHamster** [comment]
Thanks
